lib/Models/Catalog/Ows/WebFeatureServiceCatalogItem.ts

```diff
@@ -235,6 +238,64 @@ class GetCapabilitiesStratum extends LoadableStratum(
       return keyword === "GEOSERVER";
     }
   }
+
+  // Helper function to check if geojson output is supported (by checking GetCapabilities OutputTypes OR FeatureType OutputTypes)
+  hasJsonOutputFormat = (outputFormats: string[] | undefined) => {
+    return isDefined(
+      outputFormats?.find((format) =>
+        ["json", "JSON", "application/json"].includes(format)
+      )
+    );
+  };
+
+  // Find which GML formats are supported, choose the one most suited to Terria. If not available, default to "gml3"
+  @computed
+  get outputFormat(): string | undefined {
+    const supportsGeojson =
+      this.hasJsonOutputFormat(this.capabilities.outputTypes) ||
+      [...this.capabilitiesFeatureTypes.values()].reduce<boolean>(
+        (hasGeojson, current) =>
+          hasGeojson && this.hasJsonOutputFormat(current?.OutputFormats),
+        true
+      );
+
+    const searchValue = new RegExp(".*gml/3.1.1.*|.*gml3.1.1.*");
+
+    return supportsGeojson
+      ? "JSON"
+      : this.capabilities.outputTypes?.find((outputFormat) =>
+          searchValue.test(outputFormat)
+        ) ?? "gml3";
+  }
+
+  // Finds the best srsName to use. First checks if one provided in url.
+  // Then checks getCapabilities response and returns the first listed srs that is included in our list of supported srs.
+  // This enables us to use a urn identifier if supported, or a normal EPSG code if not.
+  // e.g. "urn:ogc:def:crs:EPSG::4326" or "EPSG:4326"
+  @computed
+  get srsName(): string | undefined {
+    // First check to see if URL has CRS or SRS
+    const supportedCrs = [...SUPPORTED_CRS_3857, ...SUPPORTED_CRS_4326];
+    const queryParams: any = this.catalogItem.uri?.query(true) ?? {};
+    const urlCrs =
+      queryParams.srsName ??
+      queryParams.crs ??
+      queryParams.CRS ??
+      queryParams.srs ??
+      queryParams.SRS;
+    if (urlCrs && supportedCrs.includes(urlCrs)) return urlCrs;
```

Comment:
Do we need to check if this user specified `srsName` is supported by the WFS in the getCapabilities?
Otherwise, if we try adding a url where its not supported we get an error, but it's not very descriptive.

E.g. try to add this through "Add Web Data" : 
https://webmap.yvw.com.au/YVWassets/service.svc/get?request=GetFeature&service=WFS&typeName=yvw:NDWHYDRANTS&version=1.1.0&srsName=EPSG:4283

<img width="701" alt="Screen Shot 2022-09-21 at 8 09 58 am" src="https://user-images.githubusercontent.com/7980991/191386281-333f7b36-2242-4d51-965c-7b1c3ba795cf.png">


Problem here is that WFS services dont necessarily specify ALL the srsNames they support in the getCapabilities response. I am starting to see some downsides to WFS!

Reply:
Yep I agree. If a `srsName` is provided by the user (either through URL or JSON catalog file), we should just use it even if it isn't advertised.